modules/tasks/tasks/src/main/java/org/motechproject/tasks/util/SchedulerTaskTriggerUtil.java

```diff
@@ -0,0 +1,203 @@
+package org.motechproject.tasks.util;
+
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.motechproject.commons.date.model.DayOfWeek;
+import org.motechproject.commons.date.model.Time;
+import org.motechproject.commons.date.util.DateUtil;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.scheduler.contract.CronJobId;
+import org.motechproject.scheduler.contract.CronSchedulableJob;
+import org.motechproject.scheduler.contract.DayOfWeekSchedulableJob;
+import org.motechproject.scheduler.contract.JobId;
+import org.motechproject.scheduler.contract.RepeatingJobId;
+import org.motechproject.scheduler.contract.RepeatingPeriodJobId;
+import org.motechproject.scheduler.contract.RepeatingPeriodSchedulableJob;
+import org.motechproject.scheduler.contract.RepeatingSchedulableJob;
+import org.motechproject.scheduler.contract.RunOnceJobId;
+import org.motechproject.scheduler.contract.RunOnceSchedulableJob;
+import org.motechproject.scheduler.exception.MotechSchedulerException;
+import org.motechproject.scheduler.service.MotechSchedulerService;
+import org.motechproject.tasks.domain.SchedulerJobType;
+import org.motechproject.tasks.domain.SchedulerTaskTriggerInformation;
+import org.motechproject.tasks.domain.Task;
+import org.motechproject.tasks.service.TaskService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This util class is responsible for the handling of scheduling, unscheduling and building
+ * scheduler jobs for Scheduler task triggers. Since the Scheduler trigger works differently to
+ * all other triggers, as it requires firing a task every specified amount of time, this util
+ * class has been created to encapsulate all the required logic in that area.
+ */
+@Component
+public class SchedulerTaskTriggerUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchedulerTaskTriggerUtil.class);
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormat.forPattern("yy-MM-dd HH:mm Z");
+
+    @Autowired
+    private MotechSchedulerService schedulerService;
+    @Autowired
+    private TaskService taskService;
```

Comment:
Isn't the task service already using this component? Cyclic dependency, bad design. This component should not use the task service.